src/handlers/main.ts

```diff
@@ -49,25 +53,41 @@ export default async function handlePage(
     return engines[engine](window);
   }
 
-  return fallback[urlObj.host]?.(window) || fallback["*"](window);
+  for (let match of fallback) {
+    if (micromatch.isMatch(urlObj.hostname, match.pattern)) {
+      return match.engine(window);
+    }
+  }
+
+  return engines.readability(window);
 }
 
 interface Engines {
   [key: string]: EngineFunction;
 }
 
-type EngineFunction = (window: DOMWindow) => Promise<IHandlerOutput>;
-
 export const engines: Engines = {
   readability,
   google,
   stackoverflow,
 };
 
+type EngineFunction = (window: DOMWindow) => Promise<IHandlerOutput>;
+export type EngineMatch = {
+  pattern: string | string[];
+  engine: EngineFunction;
+};
+export type EnginesMatch = EngineMatch[];
+
 export const engineList: string[] = Object.keys(engines);
 
-const fallback: Engines = {
-  "stackoverflow.com": engines.stackoverflow,
-  "www.google.com": engines.google,
-  "*": engines.readability,
-};
+export const fallback: EnginesMatch = [
```

Comment:
I'd suggest to keep it as a `[key: string]: EngineFunction` object, but okay, the array of EngineMatch looks good.